crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -180,6 +180,17 @@ def transition(inst, name, state)
         xs <=> ys
       end
 
+      # Make sure pacemaker is the first to execute on node delete
+      if state == "delete"
+        roles.each_with_index do |role, index|
```

Comment:
This loop will not work as expected. See https://stackoverflow.com/questions/31294630/is-it-safe-to-delete-from-an-array-inside-each for details.
TLDR
"when you delete an element, the elements following it will be shifted, hence the element that was supposed to be iterated next would be moved to the position of the deleted element, which has been iterated over already, and will be skipped."

Reply:
I understand, but the iteration breaks after it is deleted, ie. no further iteration after unshift. I could do:

```
prole = roles.delete_at(index)
roles.unshift(prole)
```

though I already tested it, it should be OK. Test results are in bsc#1170479.